src/Widgets/TaskDetailRevealer.vala

```diff
@@ -0,0 +1,143 @@
+
+        unowned Gtk.StyleContext due_label_context = due_label.get_style_context ();
+        due_label_context.add_provider (taskrow_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+        due_label_context.add_class ("due-date");
+
+        due_label_revealer = new Gtk.Revealer ();
+        due_label_revealer.transition_type = Gtk.RevealerTransitionType.SLIDE_RIGHT;
+        due_label_revealer.add (due_label);
+
+        description_label = new Gtk.Label (null);
+        description_label.xalign = 0;
+        description_label.lines = 1;
+        description_label.ellipsize = Pango.EllipsizeMode.END;
+        description_label.get_style_context ().add_class (Gtk.STYLE_CLASS_DIM_LABEL);
+
+        description_label_revealer = new Gtk.Revealer ();
+        description_label_revealer.reveal_child = false;
+        description_label_revealer.add (description_label);
+
+        grid = new Gtk.Grid ();
+        grid.add (due_label_revealer);
+        grid.add (description_label_revealer);
+
+        reveal_child = false;
+        add (grid);
+
+        notify["task"].connect (update_request);
+        update_request ();
+    }
+
+    private void update_request () {
+        if (task == null) {
+            reveal_child = false;
+            due_label_revealer.reveal_child = false;
+            description_label_revealer.reveal_child = false;
+
+        } else {
+            unowned ICal.Component ical_task = task.get_icalcomponent ();
+            var completed = ical_task.get_status () == ICal.PropertyStatus.COMPLETED;
+
+            if ( ical_task.get_due ().is_null_time () ) {
+                due_label_revealer.reveal_child = false;
+            } else {
+                var due_date_time = Util.ical_to_date_time (ical_task.get_due ());
+                var h24_settings = new GLib.Settings ("org.gnome.desktop.interface");
+                var format = h24_settings.get_string ("clock-format");
+
+                due_label.label = Granite.DateTime.get_relative_datetime (due_date_time);
+                due_label.tooltip_text = _("%s at %s").printf (
+                    due_date_time.format (Granite.DateTime.get_default_date_format (true)),
+                    due_date_time.format (Granite.DateTime.get_default_time_format (format.contains ("12h")))
+                );
+
+                var today = new GLib.DateTime.now_local ();
+                if (today.compare (due_date_time) > 0 && !completed) {
+                    get_style_context ().add_class ("past-due");
+                } else {
+                    get_style_context ().remove_class ("past-due");
+                }
+
+                due_label_revealer.reveal_child = true;
+            }
+
+            if (ical_task.get_description () == null) {
+                description_label_revealer.reveal_child = false;
+
+            } else {
+                var description = ical_task.get_description ();
+                description = description.replace ("\r", "").strip ();
```

Comment:
I don't like what's happening to the description, it should remain unprocessed

Reply:
The processing of the task description is becaues in "view-mode" the description is rendered to display on a single line as discussed [here](https://github.com/elementary/tasks/issues/35#issuecomment-543216958).

This idea origins of how the Apple Reminder rendered the task description prior to macOS Catalina.

Is there a better way to do this?

Reply:
I've never used an Apple computer so I might need to search for screenshots as I don't understand what's the desired effect here.

Reply:
A cleaner way would be to have a utility function (in a Utils.vala file?) that would return a wrapped string.
It would probably be more efficient to use a GLib.Regex to find an replace `(^\s+|\s+$|\n|\s\s+)` with a space :slightly_smiling_face:

Reply:
*haha* [originally wanted to use Regex](https://github.com/elementary/tasks/pull/39), but did not figure out how 🤭 thanks for the hint!

Reply:
Done using GLib.Regex.